tasks/resolve_reference.rb

```diff
@@ -44,26 +44,41 @@ def resolve_reference(opts)
   # or from an evironment variable
   def credentials(opts)
     # Ensure a credentials file was specified
-    unless opts[:credentials] || ENV[CREDENTIALS_ENV]
+    manual_creds_provided = opts[:gcloud_client_email] && opts[:gcloud_token_uri] && opts[:gcloud_private_key]
+    unless opts[:credentials] || manual_creds_provided || ENV[CREDENTIALS_ENV]
       msg = "Missing application credentials. Specify the path to the application credentials file "\
             "under the 'credentials' configuration option or as the 'GOOGLE_APPLICATION_CREDENTIALS' "\
-            "environment variable."
+            "environment variable. Credentials can also be specified using the 'gcloud_client_email', "\
+            "'gcloud_token_uri' and 'gcloud_private_key' configuration options."
       raise TaskHelper::Error.new(msg, 'bolt.plugin/validation-error')
     end
 
-    path        = File.expand_path(opts[:credentials] || ENV[CREDENTIALS_ENV], opts[:_boltdir])
-    credentials = JSON.parse(File.read(path))
-
-    # Ensure the credentials are a hash
-    unless credentials.is_a? Hash
-      msg = "Expected credentials to be a Hash, received #{credentials.class}"
-      raise TaskHelper::Error.new(msg, 'bolt.plugin/validation-error')
-    end
-
-    # Ensure the credentials have the required keys
-    if (keys = CREDENTIALS_KEYS - credentials.keys).any?
-      msg = "Missing required keys in credentials file: #{keys.join(', ')}"
-      raise TaskHelper::Error.new(msg, 'bolt.plugin/validation-error')
+    if opts[:credentials] || ENV[CREDENTIALS_ENV]
+      # Read credential hash from credential file
+      path        = File.expand_path(opts[:credentials] || ENV[CREDENTIALS_ENV], opts[:_boltdir])
+      credentials = JSON.parse(File.read(path))
+
+      # Ensure the credentials are a hash
+      unless credentials.is_a? Hash
+        msg = "Expected credentials to be a Hash, received #{credentials.class}"
+        raise TaskHelper::Error.new(msg, 'bolt.plugin/validation-error')
+      end
+
+      # Ensure the credentials have the required keys
+      if (keys = CREDENTIALS_KEYS - credentials.keys).any?
+        msg = "Missing required keys in credentials file: #{keys.join(', ')}"
+        raise TaskHelper::Error.new(msg, 'bolt.plugin/validation-error')
+      end
+    else
+      # Build credential hash from provided credential information
+      unless opts[:gcloud_client_email] && opts[:gcloud_token_uri] && opts[:gcloud_private_key]
+        msg = "Missing required keys from the credentials fields, 'gcloud_client_email', 'gcloud_private_key' "\
+              "and 'gcloud_token_uri' must be specified."
+        raise TaskHelper::Error.new(msg, 'bolt.plugin/validation-error')
+      end
```

Comment:
```suggestion
```

This is dead code since the task will already have raised an error if all three parameters weren't specified.